lib/google_sign_in_desktop.dart

```diff
@@ -458,6 +480,79 @@ class GoogleSignInDesktop extends GoogleSignInPlatform {
       await server.close();
     }
   }
+
+  @override
+  Future<bool> requestScopes(List<String> scopes) async {
+    if (scopes.isEmpty) {
+      throw ArgumentError.value(
+        scopes,
+        'scopes',
+        'Scopes must not be empty',
+      );
+    }
+
+    // Keep track of the previous scopes in case we need to revert
+    final previousScopes = _scopes;
+
+    try {
+      var tokenData = await _tokenDataStore.get();
+      if (tokenData == null) {
+        // Merge the current scopes with the requested scopes.
+        _scopes = {..._scopes, ...scopes}.toList();
+        tokenData = (await _signIn()).$1;
+
+        // It's a fresh sign in, so directly return if the requested scopes are granted
+        return _hasGrantedAllScopes(
+          tokenData: tokenData,
+          scopes: scopes,
+        );
+      }
+
+      if (_hasGrantedAllScopes(
+        tokenData: tokenData,
+        scopes: scopes,
+      )) {
+        return true;
+      }
+
+      // The requested scopes are not granted, so try to sign in with the new scopes
+      // Merge the current scopes with the requested scopes.
+      _scopes = {..._scopes, ...scopes}.toList();
+      tokenData = (await _signIn()).$1;
+
+      return _hasGrantedAllScopes(
+        tokenData: tokenData,
+        scopes: scopes,
+      );
+    } catch (_) {
+      // If something goes wrong, revert the scopes back to the previous state
+      _scopes = previousScopes;
+      rethrow;
+    }
+  }
+
+  bool _hasGrantedAllScopes({
+    required GoogleSignInDesktopTokenData tokenData,
+    required List<String> scopes,
+  }) {
```

Comment:
Could we use positional parameters instead of named parameters as both parameters are required?

```suggestion
  bool _hasGrantedAllScopes(
    GoogleSignInDesktopTokenData tokenData,
    List<String> scopes,
) {
```